storage/innobase/buf/buf0lru.cc

```diff
@@ -308,6 +308,8 @@ buf_block_t* buf_LRU_get_free_only()
 		block = reinterpret_cast<buf_block_t*>(
 			UT_LIST_GET_FIRST(buf_pool.free));
 	}
+	if (block)
+		block->page.set_os_used();
 
 	return(block);
```

Comment:
To avoid adding a conditional branch, the call should better be added right before the `break` inside the `while (bpage != NULL)` loop.

Reply:
It looks like the last `block = reinterpret_cast ...` in the loop could also set the `block` pointer and moving this condition would mean `set_os_used()` isn't called.